etna/models/nn/timesfm.py

```diff
@@ -0,0 +1,339 @@
+        NotImplementedError:
+            if forecasting is done without exogenous features and dataset has None frequency.
+        """
+        if return_components:
+            raise NotImplementedError("This mode isn't currently implemented!")
+
+        max_context_size = len(ts.index) - prediction_size
+        if max_context_size <= 0:
+            raise ValueError("Dataset doesn't have any context timestamps.")
+
+        if max_context_size < self.context_size:
+            warnings.warn("Actual length of a dataset is less that context size. All history will be used as context.")
+
+        self.tfm._set_horizon(prediction_size)
+
+        end_idx = len(ts.index)
+
+        static_reals_dict = (
+            {column: ts.df.loc[ts.index[0], pd.IndexSlice[:, column]].values.tolist() for column in self.static_reals}
+            if self.static_reals is not None
+            else None
+        )
+        static_categoricals_dict = (
+            {
+                column: ts.df.loc[ts.index[0], pd.IndexSlice[:, column]].values.tolist()
+                for column in self.static_categoricals
+            }
+            if self.static_categoricals is not None
+            else None
+        )
+        time_varying_reals_dict = (
+            {
+                column: ts.df.loc[:, pd.IndexSlice[:, column]].values.swapaxes(1, 0).tolist()
+                for column in self.time_varying_reals
+            }
+            if self.time_varying_reals is not None
+            else None
+        )
+        time_varying_categoricals_dict = (
+            {
+                column: ts.df.loc[:, pd.IndexSlice[:, column]].values.swapaxes(1, 0).tolist()
+                for column in self.time_varying_categoricals
+            }
+            if self.time_varying_categoricals is not None
+            else None
+        )
+
+        future_ts = ts.tsdataset_idx_slice(start_idx=end_idx - prediction_size, end_idx=end_idx)
+
+        if static_reals_dict or static_categoricals_dict or time_varying_reals_dict or time_varying_categoricals_dict:
+            target = ts.df.loc[:, pd.IndexSlice[:, "target"]].dropna().values.swapaxes(1, 0).tolist()
+
+            complex_forecast, _ = self.tfm.forecast_with_covariates(
+                inputs=target,
+                dynamic_numerical_covariates=time_varying_reals_dict,
+                dynamic_categorical_covariates=time_varying_categoricals_dict,
+                static_numerical_covariates=static_reals_dict,
+                static_categorical_covariates=static_categoricals_dict,
+                freq=[freq_map(ts.freq)] * len(ts.segments),
+            )
+            future_ts.df.loc[:, pd.IndexSlice[:, "target"]] = np.vstack(complex_forecast).swapaxes(1, 0)
+        else:
+            if ts.freq is None:
+                raise NotImplementedError(
+                    "Data with None frequency isn't currently implemented for forecasting without exogenous features."
```

Comment:
Smth is wrong with this phrase. It looks like the data isn't implemented, not the forecasting for such data.